packages/flutter_tools/lib/src/ios/mac.dart

```diff
@@ -160,6 +171,8 @@ class IMobileDevice {
       );
       if (result.exitCode == 255 && result.stdout != null && result.stdout.contains('No device found'))
         throw IOSDeviceNotFoundError('ideviceinfo could not find device:\n${result.stdout}');
+      if (result.exitCode == 255 && result.stderr != null && result.stderr.contains('Could not connect to lockdownd'))
+        throw IOSDeviceNotTrustedError('Device info unavailable until the user selects "Trust This Computer" on the device (${result.stderr.trim()})');
```

Comment:
The problem with this is that there are other cases where ideviceinfo would return "Could not connect to lockdownd" that may require different action...i.e. the error code is not irrelevant. See this long enum: https://github.com/libimobiledevice/libimobiledevice/blob/26373b334889f5ae2e2737ff447eb25b1700fa2f/include/libimobiledevice/lockdown.h#L59

These non-"Trust This Computer" cases seem rare for flutter developers, but I wasn't sure if we should just bucket them ALL together.

Reply:
How would a different sort of message sound? Like:
'Device info unavailable. Is the device asking to "Trust this Computer"?'